File: Example Project/Test/ViewController4.m
//
//  ViewController4.m
//  Test
//
//  Created by Prince Ugwuh on 8/10/16.
//  Copyright © 2016 TTT. All rights reserved.
//

#import "ViewController4.h"
#import "Test-Swift.h"
#import "TestTextFieldCell.h"

@interface ViewController4 () <MSTableViewDataSourceDelegate>
@property(nonatomic, weak, nullable) IBOutlet UITableView *tableView;
@property(nonatomic, strong, nullable) MSTableViewDataSource *dataSource;
@end

@implementation ViewController4

- (void)viewDidLoad {
    [super viewDidLoad];
    
    [self.tableView registerNib:[UINib nibWithNibName:[TestTextFieldCell identifier] bundle:nil] forCellReuseIdentifier:[TestTextFieldCell identifier]];
    self.tableView.estimatedRowHeight = 54;
    self.tableView.rowHeight = UITableViewAutomaticDimension;
    
    [self build];
}


- (void)build {
    
    
    MSTableViewDataSourceCellItem *item = [[MSTableViewDataSourceCellItem alloc] init];
    item.cellIdentifier = [TestTextFieldCell identifier];
    item.item = nil;
    
    self.dataSource = [[MSTableViewDataSource alloc] initWithCellItems:@[item]];
    self.dataSource.delegate = self;
    
    self.tableView.dataSource = self.dataSource;
    
}

- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}

// Mark - MSTableViewDataSourceDelegate

- (void)dataSource:(TestTextFieldCell *)cell didConfigureCellAtIndexPath:(NSIndexPath * _Nonnull)indexPath {
    cell.textFieldValueDidChange = ^(NSString *text){
        NSLog(@"%@", text);
    };
}

@end

File: Example Project Swift/Example Project Swift/ViewController4.swift
//
//  ViewController4.swift
//  Example Project Swift
//
//  Created by Prince Ugwuh on 10/12/16.
//  Copyright © 2016 MFX Studios. All rights reserved.
//

import UIKit

class ViewController4: UIViewController, TableViewDataSourceDelegate {
    @IBOutlet weak var tableView: UITableView!
    
    var dataSource: TableViewDataSource?
    
    override func viewDidLoad() {
        super.viewDidLoad()
        
        self.tableView.register(UINib(nibName: TestTextFieldCell.identifier, bundle: nil), forCellReuseIdentifier: TestTextFieldCell.identifier)
        self.tableView.estimatedRowHeight = 54
        self.tableView.rowHeight = UITableViewAutomaticDimension
        
        build()
    }
    
    func build() {
        
        let item = TableViewDataSourceCellItem()
        item.cellIdentifier = TestTextFieldCell.identifier
        
        self.dataSource = TableViewDataSource(cellItems: [item])
        self.dataSource?.delegate = self
        self.tableView?.dataSource = self.dataSource
        
    }
    
    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }
    
    func didConfigureCell(_ cell: TableViewCellDataSource, atIndexPath indexPath: IndexPath) {
        let textFieldCell = cell as? TestTextFieldCell
        textFieldCell?.valueDidChange = { text in
            print(text)
        }
    }
}
